src/transformers/pipelines/table_question_answering.py

```diff
@@ -235,52 +238,76 @@ def __call__(self, *args, **kwargs):
-            if not query:
-                raise ValueError("query is empty")
-            inputs = self.tokenizer(
-                table, query, return_tensors=self.framework, truncation="drop_rows_to_fit", padding=padding
-            )
-
-            outputs = self.sequential_inference(**inputs) if sequential else self.batch_inference(**inputs)
-
-            if self.aggregate:
-                logits, logits_agg = outputs[:2]
-                predictions = self.tokenizer.convert_logits_to_predictions(inputs, logits.detach(), logits_agg)
-                answer_coordinates_batch, agg_predictions = predictions
-                aggregators = {i: self.model.config.aggregation_labels[pred] for i, pred in enumerate(agg_predictions)}
-
-                no_agg_label_index = self.model.config.no_aggregation_label_index
-                aggregators_prefix = {
-                    i: aggregators[i] + " > " for i, pred in enumerate(agg_predictions) if pred != no_agg_label_index
-                }
-            else:
-                logits = outputs[0]
-                predictions = self.tokenizer.convert_logits_to_predictions(inputs, logits.detach())
-                answer_coordinates_batch = predictions[0]
-                aggregators = {}
-                aggregators_prefix = {}
-
-            answers = []
-            for index, coordinates in enumerate(answer_coordinates_batch):
-                cells = [table.iat[coordinate] for coordinate in coordinates]
-                aggregator = aggregators.get(index, "")
-                aggregator_prefix = aggregators_prefix.get(index, "")
-                answer = {
-                    "answer": aggregator_prefix + ", ".join(cells),
-                    "coordinates": coordinates,
-                    "cells": [table.iat[coordinate] for coordinate in coordinates],
-                }
-                if aggregator:
-                    answer["aggregator"] = aggregator
-
-                answers.append(answer)
-            if len(answer) == 0:
-                raise PipelineException("Empty answer")
-            batched_answers.append(answers if len(answers) > 1 else answers[0])
-        return batched_answers if len(batched_answers) > 1 else batched_answers[0]
+        pipeline_inputs = self._args_parser(*args, **kwargs)
+
+        results = super().__call__(pipeline_inputs, **kwargs)
+        if len(results) == 1:
+            return results[0]
+        return results
+
+    def set_parameters(self, sequential=None, padding=None, truncation=None, **kwargs):
+        if sequential is not None:
+            self.sequential = sequential
+        if padding is not None:
+            self.padding = padding
+        if truncation is not None:
+            self.truncation = truncation
+
+    def preprocess(self, pipeline_input):
+        table, query = pipeline_input["table"], pipeline_input["query"]
+        if table.empty:
+            raise ValueError("table is empty")
+        if not query:
```

Comment:
```suggestion
        if query is None:
```

Reply:
I think we can't change because `query == ""` is possible at this point.
Maybe we could add a test for this case (just making sure it does not go away later in the future)

Reply:
In this case, it should explicitely test both:
```
if query is None or len(query) == ""
```
Relying on Python magic to convert something to a truthy value is unreliable and can be source of bugs that are hard to track

Reply:
Overly explicit is good you're right !